diff --git a/node_modules/@dominative/solid/nativescript.webpack.js b/node_modules/@dominative/solid/nativescript.webpack.js
index 1588764..1647573 100644
--- a/node_modules/@dominative/solid/nativescript.webpack.js
+++ b/node_modules/@dominative/solid/nativescript.webpack.js
@@ -35,7 +35,7 @@ const solid = (config, env) => {
     .end()
     .use('babel-loader')
     .loader('babel-loader')
-    .before('ts-loader')
+    .before("ts-loader")
     .options({
       babelrc: false,
       configFile: false,
@@ -47,6 +47,7 @@ const solid = (config, env) => {
             generate: 'universal',
           },
         ],
+        "@babel/typescript"
       ],
       env: {
         development: {
